File: services/identity-service/internal/logger/logger.go
package logger

import "go.uber.org/zap"

var log *zap.Logger

func Init() error {
	var err error
	log, err = zap.NewProduction()
	return err
}

func Info(msg string, fields ...zap.Field) {
	log.Info(msg, fields...)
}

func Error(msg string, fields ...zap.Field) {
	log.Error(msg, fields...)
}

func Fatal(msg string, fields ...zap.Field) {
	log.Fatal(msg, fields...)
}

File: services/video-service/src/utils/logger.ts
import { Logger } from 'pino';

export const logger = {
  info: (msg: string | object, meta?: any) => {
    if (typeof msg === 'string') {
      console.log(`[INFO] ${msg}`, meta || '');
    } else {
      console.log('[INFO]', msg);
    }
  },
  error: (msg: string | object, meta?: any) => {
    if (typeof msg === 'string') {
      console.error(`[ERROR] ${msg}`, meta || '');
    } else {
      console.error('[ERROR]', msg);
    }
  },
  warn: (msg: string | object, meta?: any) => {
    if (typeof msg === 'string') {
      console.warn(`[WARN] ${msg}`, meta || '');
    } else {
      console.warn('[WARN]', msg);
    }
  },
  debug: (msg: string | object, meta?: any) => {
    if (typeof msg === 'string') {
      console.debug(`[DEBUG] ${msg}`, meta || '');
    } else {
      console.debug('[DEBUG]', msg);
    }
  },
};
